### PR TITLE
1168 feature add authz unit tests to ci

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
# PR Description

# Dependencies

#1159  should be merged first.

# Tasks

* [x] - A description of the PR has been provided, and a diagram included if it is a new feature.
* [x] - Formatter has been run
* [x] - CHANGELOG comment has been added
* [x] - Labels have been assigned to the pr
* [x] - A reviwer has been added
* [x] - A user has been assigned to work on the pr
* [x] - If new feature a unit test has been added

## Summary by Sourcery

Add unit tests for the AuthzWorker class and integrate them into the CI pipeline. Refactor the AuthzWorker class to enhance path validation and authorization logic. Introduce a new CI job to run authorization unit tests using the GCS production container.

New Features:
- Introduce the AuthzWorker class to handle authorization checking, path validation, and FTP URL processing.

Enhancements:
- Refactor the AuthzWorker class to improve path validation and authorization logic.

CI:
- Add a new CI job to run authorization unit tests using the same container as the GCS production container.

Tests:
- Add unit tests for the AuthzWorker class to validate path and URL handling, authorization checks, and response processing.
- Introduce unit tests for the getVersion function to verify version retrieval.

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
Screenshot (220)

This shows where the CI job for testing authz has been added in the CI pipeline, it has been added to the unit testing stage.

**Comment by JoshuaSBrown** [comment]
@par-hermes format

**Review by AronPerez — CHANGES_REQUESTED** [review]

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+class ConfigFixture {
+public:
+  // Setup: create the object before each test case
+  ConfigFixture() {
+    std::string repo_id = "datafed-one-repo-to-rule-them-all";
+    std::string server_addr = "tcp://ruler:7513";
+    std::string pub_key = "=+r^&Qc&}f<Pho7ViJwVM5(Ze^o%UnnyA:XjF}7Q";
+    std::string priv_key = "X4y+AFXCmBoyKrc0J0V5}Z2HpCYRf3(wvhe**z))";
+    std::string server_key = "a.r)OeoL=?/rHaK1-ow<xOd7YxM/6)A&Kdu]*<)3";
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:30]: Yes, this is randomly created, specifically for testing, good spot though!

**Review by nedvedba — APPROVED** [review]

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+private:
+};
+
+BOOST_FIXTURE_TEST_SUITE(AuthzTest, ConfigFixture)
+
+BOOST_AUTO_TEST_CASE(test_authz) {
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  std::cout << std::string(config.repo_id) << std::endl;
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:64]: Is the point of this test to validate an error does not occur during worker construction?

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+  std::string expected_result = "/a/b/c/d/e.txt";
+
+  BOOST_CHECK_EQUAL(worker.getAuthzPath(deep_path), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(ProcessResponseWithTimeout) {
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  SDMS::ICommunicator::Response response;
+  response.time_out = true;
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:318]: I'm curious how this timeout works in practice. Is this processing job added to a queue?

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+  response.message->setPayload(std::move(auth_by_token_req));
+
+  std::string route = "MtOlympia";
+  response.message->addRoute(route);
+
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+
+  BOOST_CHECK_EQUAL(worker.processResponse(response), 0);
+  // Verify that DL_DEBUG was called with "Received NACK reply" for NACK case
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:376]: I believe this comment should be moved to the next test.

**Review by t-ramz — APPROVED** [review]
Looks good! Just a few questions and comments.

File: repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp (hunk 2 above)
Comment (JoshuaSBrown) [reply repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp:64]: Yes, the test name could be improved.

**Review by AronPerez — APPROVED** [review]

**Pull request merged**